accord-core/src/main/java/accord/primitives/Ranges.java

```diff
@@ -227,4 +230,27 @@ else if (count == result.length)
         return construct(cachedRanges.completeAndDiscard(result, count));
     }
 
+    public Partition partition(Predicate<Range> test)
+    {
+        List<Range> left = new ArrayList<>();
+        List<Range> right = new ArrayList<>();
+        for (Range range : this)
+            (test.test(range) ? left : right).add(range);
```

Comment:
Have a better name?

Reply:
Trues and falses?

Reply:
was trying to find other APIs to see if there are better names... and best I see in JDK is

```
Map<Boolean, List<Integer>> groups = intList.stream().collect(Collectors.partitioningBy(s -> s > 6));
```

So rather than an object with true/false, this returns a `Map<Boolean, T>`

I don't mind mimicking this API if you think its best

Reply:
That WFM.